pr_reminder.rb

```diff
@@ -1,35 +1,55 @@
 #! /usr/bin/env ruby
+require 'open-uri'
+require 'json'
+require 'ostruct'
 
-require "open-uri"
-require "json"
-require "ostruct"
+# Represents an Organization in GitHub
+class Organization
+  attr_reader :name
 
-def usage
-<<-END
-  #{$0} <github organization>
-  Tool to print report about pending pull request of given organization
-END
-end
+  def initialize(name)
+    @name = name
+  end
+
+  # Retrieves all the repositories of the organization (with PRs by default)
+  def repositories_with_pull_requests(reponames = [])
+    repositories = []
+    page = 1
+    loop do
+      repositories_in_page = load_repos_in_page(page)
+      break if repositories_in_page.empty?
+      repositories.concat(repositories_in_page)
+      page += 1
+    end
+    repositories.select! { |repo| reponames.include?(repo.name) } unless reponames.empty?
+    repositories.select!(&:any_pull_requests?)
+    repositories
+  end
 
-def print_usage
-  $stdout.puts usage
+  private
+
+  # Load repositories in the page from GitHub
+  def load_repos_in_page(page)
+    url = "https://api.github.com/orgs/#{name}/repos?page=#{page}&per_page=100"
+    open(url) do |json_file|
+      JSON.load(json_file.gets).map { |repo_data| Repository.new(repo_data) }
+    end
+  end
 end
 
+# Represents a Repository in GitHub
 class Repository < OpenStruct
-  def self.all(organization)
-    result = []
-    i = 1
-    cont = true
-    begin
-      tmp = load_part(organization, i)
-      if tmp.empty?
-        cont = false
-      else
-        result.concat tmp
-      end
-      i += 1
-    end while (cont)
-    return result
+  def pending_pull_requests
+    pull_requests = []
+    url = "https://api.github.com/repos/#{full_name}/pulls"
+    open(url) do |json_file|
+      pull_requests = JSON.load(json_file.gets).map { |pr_data| PullRequest.new(pr_data) }
+    end
+    pull_requests.select! do |pull_request|
+      days = (Time.now.monday? || Time.now.tuesday?) ? 5 : 3 # do not count weekend in pending time
+      pull_request.pending_days > days
+    end
+    pull_requests
```

Comment:
if you do not use `select!` and instead use `select` then this line is not needed as result of select is what we want.